### ISSUE TITLE
Questions on sync bn implementation

### ISSUE THREAD

**Issue by chengmengli06** [NVIDIA/apex#653]
For https://github.com/NVIDIA/apex/blob/master/apex/parallel/sync_batchnorm.py#L85:
           if not self.training and self.track_running_stats:
  Should it be  if not (self.training and self.track_running_stats) ?

For https://github.com/NVIDIA/apex/blob/master/apex/parallel/sync_batchnorm.py#L133:
   It seems that there are gradients on mean, var, which are required to calculate the gradients on input.

**Comment by chengmengli06** [comment]
@mcarilli could you have a look at my question?
